packages/common/src/directives/ng_optimized_image/lcp_image_observer.ts

```diff
@@ -98,20 +97,20 @@ export class LCPImageObserver implements OnDestroy {
       alreadyWarnedModified: false,
       alreadyWarnedPriority: false,
     };
-    this.images.set(getUrl(rewrittenSrc, this.window!).href, newObservedImageState);
+    this.images.set(getUrl(rewrittenSrc, window).href, newObservedImageState);
   }
 
   unregisterImage(rewrittenSrc: string) {
     if (!this.observer) return;
-    this.images.delete(getUrl(rewrittenSrc, this.window!).href);
+    this.images.delete(getUrl(rewrittenSrc, window).href);
   }
 
   updateImage(originalSrc: string, newSrc: string) {
```

Comment:
It'd be great to add an exit condition into this function as well:
```suggestion
  updateImage(originalSrc: string, newSrc: string) {
    if (!this.observer) return;
```